R/html-tweak.R

```diff
@@ -129,6 +129,113 @@ tweak_footnotes <- function(html) {
   xml2::xml_remove(container)
 }
 
+# Tabsets tweaking: find Markdown recommended in https://bookdown.org/yihui/rmarkdown-cookbook/html-tabs.html
```

Comment:
For the record, in rmarkdown the transformation happens via JS https://github.com/rstudio/rmarkdown/blob/145046175c721af185aa6ba3ecc9262f12dc7369/inst/rmd/h/navigation-1.1/tabsets.js#L50

Reply:
We are currently discussing a bit tabset feature because of https://github.com/rstudio/rmarkdown/pull/2054. We got input from @cpsievert that bslib will offer functions for creating navs / tabs UI (https://rstudio.github.io/bslib/reference/index.html#section-create-navs-and-navbars) and that we may be able to leverage that in some way in the future for rmarkdown.

Until now our direction for tabset feature would be to use Lua filters rather than R directly. Also because Quarto has already a Lua filter for this (but using a fenced div syntax). But nothing is planned and started on this. 

I think Lua filter would not help at all pkgdown usage currently, am I right ? 

If we don't go the Lua filter road, then yes regarding xml2, I think it could be included in the Rmd ecosystem. it has no dependency and I feel we should use it more for some html post processing. But it is not already in the package tree for rmarkdown.

Reply:
If users use bslib for creating navs, then the resulting divs should be protected i.e. **not** treated by pkgdown. The code in this PR recognizes future tabsets via the XPath query `.//div[contains(@class, 'tabset')]` which if I follow correctly would not pick up navs created by bslib. :thinking: